### ISSUE TITLE
Need validation test to catch cube.c's invalid use of swapchain images

### ISSUE THREAD

**Issue by courtney-g** [KhronosGroup/Vulkan-LoaderAndValidationLayers#412]
Issue #76 from Vulkan-Docs repo. (https://github.com/LunarG/VulkanSamples/issues/76#issuecomment-214224620)

Need validation test for this spec language:
Use of a presentable image must occur only after the image is returned by vkAcquireNextImageKHR, and before it is presented by vkQueuePresentKHR. This includes transitioning the image layout and rendering commands

**Comment by karl-lunarg** [comment]
@ianelliottus , is this still an active issue?  If so, can you suggest next steps?

**Comment by chrisforbes** [comment]
Being able to validate this properly is one of the things that will fall out of my swapchain -> core_validation merge.

**Comment by ianelliottus** [comment]
@karl-lunarg, yes, this is still an active issue.  This is a pretty-difficult thing to check for.  The work that @chrisforbes is doing of migrating the swapchain validation layer into core_validation should make it possible to do this.  It makes sense to assign this to somebody besides me, as the checks will be outside of the WSI commands (e.g. if you're rendering to an image, and if it's a swapchain image, was it acquired, but not presented?).

**Comment by ratchetfreak** [comment]
I think you can leverage the existing read/write hazard checking.

Swapchain image start out being read by the presentation engine, after acquire call the sync point is the semaphore/fence until they are available. Then it needs a sync when passing for present. 

The only real trickyness is that vkDeviceWaitIdle doesn't affect the being read by presentation engine state

**Comment by karl-lunarg** [comment]
OK, thanks @chrisforbes for taking this on.

**Comment by mark-lunarg** [comment]
@chrisforbes, is this one baked?

**Comment by chrisforbes** [comment]
Not fully. The swapchain -> CV merge was completed some time ago, and we track the acquired bit now on images, but we don't complain about misuse.

**Comment by karl-lunarg** [comment]
@chrisforbes I took a crack at this - can you please take a look and let me know if this is the right direction and if I'm missing anything?

I still need to write a test.

The cube app has changed a lot since this was opened.  The layout transitions now are requested in the render pass.   But I was still able to exercise this check by hacking cube to do a layout transition in the "prepare buffers" step, which submits the barrier commands without acquiring the images.
